File: autowiring/src/main/kotlin/com/mthaler/bankapp/dao/AccountStatementDaoImpl.java
package com.mthaler.bankapp.dao;

import java.util.Date;

import com.mthaler.bankapp.domain.AccountStatement;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class AccountStatementDaoImpl implements AccountStatementDao {

	private static Logger logger = LogManager.getLogger(AccountStatementDaoImpl.class);
	
	@Override
	public AccountStatement getAccountStatement(Date from, Date to) {
		logger.info("Getting account statement");
		return new AccountStatement();
	}

}

File: autowiring/src/main/kotlin/com/mthaler/bankapp/dao/AccountStatementDaoImpl.kt
package com.mthaler.bankapp.dao

import com.mthaler.bankapp.dao.AccountStatementDao
import com.mthaler.bankapp.domain.AccountStatement
import com.mthaler.bankapp.dao.AccountStatementDaoImpl
import org.apache.logging.log4j.LogManager
import java.util.*

class AccountStatementDaoImpl : AccountStatementDao {

    override fun getAccountStatement(from: Date?, to: Date?): AccountStatement? {
        logger.info("Getting account statement")
        return AccountStatement()
    }

    companion object {
        private val logger = LogManager.getLogger(
            AccountStatementDaoImpl::class.java
        )
    }
}